pulp_python/tests/functional/api/test_pypi_apis.py

```diff
@@ -26,154 +19,202 @@
-        )
 
-    def test_package_upload_session(self):
-        """Tests that multiple uploads will be broken up into multiple tasks."""
-        repo, distro = self._create_empty_repo_and_distribution()
-        url = urljoin(PYPI_HOST, distro.base_path + "/legacy/")
-        session = requests.Session()
-        response = session.post(
-            url,
-            data={"sha256_digest": PYTHON_EGG_SHA256},
-            files={"content": open(self.egg, "rb")},
-        )
-        self.assertEqual(response.status_code, 200)
-        response = response.json()
-        monitor_task(response["task"])
-        response2 = session.post(
-            url,
-            data={"sha256_digest": PYTHON_WHEEL_SHA256},
-            files={"content": open(self.wheel, "rb")},
-        )
-        self.assertEqual(response2.status_code, 200)
-        response2 = response2.json()
-        self.assertNotEqual(response["task"], response2["task"])
-        monitor_task(response2["task"])
-        content = get_content_summary(repo, f"{repo.versions_href}2/")
-        self.assertDictEqual({PYTHON_CONTENT_NAME: 2}, content)
-
-    def test_package_upload_simple(self):
-        """Tests that the package upload endpoint exposed at `/simple/` works."""
-        repo, distro = self._create_empty_repo_and_distribution()
-        url = urljoin(PYPI_HOST, distro.base_path + "/simple/")
-        response = requests.post(
+
+@pytest.fixture
+def python_empty_repo_distro(python_repo_factory, python_distribution_factory):
+    """Returns an empty repo with and distribution serving it."""
+    def _generate_empty_repo_distro(repo_body=None, distro_body=None):
+        repo_body = repo_body or {}
+        distro_body = distro_body or {}
+        repo = python_repo_factory(**repo_body)
+        distro = python_distribution_factory(repository=repo, **distro_body)
+        return repo, distro
+
+    yield _generate_empty_repo_distro
+
+
+@pytest.mark.parallel
+def test_empty_index(python_bindings, python_empty_repo_distro):
+    """Checks that summary stats are 0 when index is empty."""
+    _, distro = python_empty_repo_distro()
+
+    summary = python_bindings.PypiApi.read(path=distro.base_path)
+    assert not any(summary.to_dict().values())
+
+
+@pytest.mark.parallel
+def test_live_index(
+    python_bindings, python_repo_with_sync, python_remote_factory, python_distribution_factory
+):
+    """Checks summary stats are correct for indexes pointing to repositories."""
+    remote = python_remote_factory(includes=PYTHON_MD_PROJECT_SPECIFIER)
+    repo = python_repo_with_sync(remote)
+    distro = python_distribution_factory(repository=repo)
+
+    summary = python_bindings.PypiApi.read(path=distro.base_path)
+    assert summary.to_dict() == PYTHON_MD_PYPI_SUMMARY
+
+
+@pytest.mark.parallel
+def test_published_index(
+    python_bindings,
+    python_repo_with_sync,
+    python_remote_factory,
```

Comment:
I'm beginning to wonder if we should put all the factories into the bindings fixture too.